File: play-services-core/src/main/java/org/microg/gms/appinvite/AppInviteService.java
/*
 * Copyright (C) 2019 e Foundation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.microg.gms.appinvite;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.util.Log;
import android.os.RemoteException;

import com.google.android.gms.common.api.CommonStatusCodes;
import com.google.android.gms.common.internal.GetServiceRequest;
import com.google.android.gms.common.internal.IGmsCallbacks;

import org.microg.gms.BaseService;
import org.microg.gms.common.GmsService;
import org.microg.gms.common.PackageUtils;

import org.microg.gms.appinvite.AppInviteServiceImpl;

public class AppInviteService extends BaseService {
    private static final String TAG = "GmsAppInviteService";

    public AppInviteService() {
        super("GmsAppInviteSvc", GmsService.APP_INVITE);
    }

    @Override
    public void handleServiceRequest(IGmsCallbacks callback, GetServiceRequest request, GmsService service) throws RemoteException {
        PackageUtils.getAndCheckCallingPackage(this, request.packageName);
        Log.d(TAG, "callb: " + callback + " ; req: " + request + " ; serv: " + service);

        callback.onPostInitComplete(0, new AppInviteServiceImpl(this, request.packageName, request.extras), null);
    }
}

File: play-services-appinvite/core/src/main/kotlin/org/microg/gms/appinivite/AppInviteService.kt
/*
 * SPDX-FileCopyrightText: 2019 e Foundation
 * SPDX-FileCopyrightText: 2023 microG Project Team
 * SPDX-License-Identifier: Apache-2.0
 */
package org.microg.gms.appinivite

import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.os.Parcel
import android.os.RemoteException
import android.util.Log
import com.google.android.gms.appinvite.internal.IAppInviteCallbacks
import com.google.android.gms.appinvite.internal.IAppInviteService
import com.google.android.gms.common.api.Status
import com.google.android.gms.common.internal.GetServiceRequest
import com.google.android.gms.common.internal.IGmsCallbacks
import org.microg.gms.BaseService
import org.microg.gms.common.GmsService
import org.microg.gms.common.PackageUtils
import org.microg.gms.utils.warnOnTransactionIssues

private const val TAG = "AppInviteService"

class AppInviteService : BaseService(TAG, GmsService.APP_INVITE) {
    override fun handleServiceRequest(callback: IGmsCallbacks, request: GetServiceRequest, service: GmsService) {
        PackageUtils.getAndCheckCallingPackage(this, request.packageName)
        Log.d(TAG, "callb: $callback ; req: $request ; serv: $service")
        callback.onPostInitComplete(0, AppInviteServiceImpl(this, request.packageName, request.extras), null)
    }
}


class AppInviteServiceImpl(context: Context?, packageName: String?, extras: Bundle?) : IAppInviteService.Stub() {
    override fun updateInvitationOnInstall(callback: IAppInviteCallbacks, invitationId: String) {
        callback.onStatus(Status.SUCCESS)
    }

    override fun convertInvitation(callback: IAppInviteCallbacks, invitationId: String) {
        callback.onStatus(Status.SUCCESS)
    }

    override fun getInvitation(callback: IAppInviteCallbacks) {
        callback.onStatusIntent(Status(Activity.RESULT_CANCELED), null)
    }

    override fun onTransact(code: Int, data: Parcel, reply: Parcel?, flags: Int): Boolean = warnOnTransactionIssues(code, reply, flags, TAG) { super.onTransact(code, data, reply, flags) }
}
